simcon_project/static/conversation_templates/create_conversation_template.js

```diff
@@ -0,0 +1,346 @@
+        this.nodeName = ""                  // Node's name
+        this.lastChoiceIndex = 0            // Used to keep track of the last choice index used (used as key in responseChoices map below)
+
+        // Items to be propagated ot the backend
+        this.responseChoices = new Map()    // Array of choices for this node
+        this.videoUrl = ""                  // Stores non-embeddable video url (url as user provides it)
+        this.nodeDescription = ""           // Node's description
+        this.isFirst = false                // True if node is first
+        this.isTerminal = false             // True if node is terminal
+    }
+}
+
+// Used inside StepNode to hold available choices at that step
+class Choice {
+    constructor(description, destinationIndex) {
+        this.description = description
+        this.destinationIndex = destinationIndex // Stores the node index that serves as the destination for this choice
+        // (associated with the 'value' element inside the relevant 'option' in a destination 'select' [look at choice cards])
+        // 0 if no destination selected
+    }
+}
+
+/**
+ * Called on page load. Sets the listeners and initializes page content.
+ */
+function loadState() {
+    $(document).ready()
+
+    addStepNode()
+    updateNodeInFocus(1)
+
+    // When an input is changed, change unsaved indicator to true so that the page is saved if another node is put in focus
+    $(document).on("change", ":input", function () {
+        unsaved = true
+    })
+
+    // When step node name is updated, update the state variable and name on the associated step node card
+    $(document).on("input", "#node-name-input", function () {
+        $("#step-" + currentNodeInFocus.index + " .card-title").text(getNodeNameInput())
+    })
+
+    // When a step node card is clicked, that card's context has to be focused
+    $(document).on("click", ".step-node-card", function () {
+        let clickedNodeId = parseInt($(this).attr('id').split('-')[1])
+
+        if(!(nodes.get(clickedNodeId) === currentNodeInFocus)) {
+            updateNodeInFocus(clickedNodeId)
+        }
+    })
+
+    // Every step node card has a 'Delete' link
+    // If click propagation is not stopped, the parent card's onclick event defined above is also triggered when Delete is clicked
+    $(document).on("click", ".remove-step-node", function (event) {
+        event.stopPropagation()
+    })
+
+    // When user updates the video url, updates the embedded video
+    $("#video-url-input").blur(() => {
+        setEmbeddedVideoUrl(getVideoUrlInput())
+    })
+}
+
+/**
+ * Given a node index, switches focus to it by updating the input and choice data.
+ *
+ * @param nodeIndex index of the node to be put in focus
+ */
+function updateNodeInFocus(nodeIndex) {
+    saveValuesIfNeeded()
+
+    if(currentNodeInFocus !== null) $("#step-"+currentNodeInFocus.index).css("background-color", "white")
```

Comment:
why not add the css in the static file with a tag that signifies not in focus

Reply:
Eh, could do either. Doing it this way is worth it since the other way wouldn't really clean things up, but would put things in different places.